tests/integration/questionnaire/test_questionnaire_csrf.py

```diff
@@ -1,27 +1,124 @@
+import json
+
 from tests.integration.integration_test_case import IntegrationTestCase
 
 
 class TestQuestionnaireInterstitial(IntegrationTestCase):
 
     def setUp(self):
         super().setUp()
-        self.launchSurvey('test', 'interstitial_page')
+        self.launchSurvey('test', 'interstitial_page', roles=['dumper'])
         self.assertInPage('Your response is legally required')
         self.first_url = self.last_url
 
     def test_form_not_processed_with_no_csrf_token(self):
         self.last_csrf_token = None
         self.post(action='start_questionnaire')
-        self.assertStatusOK()
+        self.assertStatusForbidden()
         self.assertEqualUrl(self.first_url)
 
-    def test_form_not_processed_with_incorrect_csrf_token(self):
+    def test_given_on_introduction_page_when_submit_invalid_token_then_forbidden(self):
         self.last_csrf_token = 'made-up-token'
         self.post(action='start_questionnaire')
-        self.assertStatusOK()
+        self.assertStatusForbidden()
         self.assertEqualUrl(self.first_url)
 
-    def test_form_processed_with_correct_csrf_token(self):
+    def test_given_on_introduction_page_when_submit_valid_token_then_redirect_to_next_page(self):
         self.post(action='start_questionnaire')
         self.assertStatusOK()
         self.assertInPage('What is your favourite breakfast food')
+
+    def test_given_answered_question_when_change_answer_with_invalid_csrf_token_then_answers_not_saved(self):
+        # Given
+        self.post()
+        post_data = {
```

Comment:
I had been in-lining these to keep the verbosity down e.g.

`self.post({'favourite-breakfast': 'Muesli'})`